### PR TITLE
Navigation: fix `bodyScrolling` in safari for browseDashboards/scenes dashboards

### LINKED ISSUE #90929: Scenes Dashboard empty page with `bodyScrolling` in Safari
Reported https://raintank-corp.slack.com/archives/C036J5B39/p1721867944029689

In macOS Safari, Dashboards (with scenes enabled) displays an empty page with the `bodyScrolling` feature toggle enabled. It appears Dashboards has its own instance of CustomScrollbars that fails to get any height, resulting in a visually empty page.

### LINKED ISSUE #90930: BrowseDashboards empty page with `bodyScrolling` in Safari
Reported https://raintank-corp.slack.com/archives/C036J5B39/p1721867944029689

In macOS Safari, BrowseDashboards displays an empty page with the bodyScrolling feature toggle enabled. I have not verified this myself.

Suspect AutoSizer fails to get full height to display anything.

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- correctly flexes pages in safari

**Why do we need this feature?**

- so pages appear correctly in safari when `bodyScrolling` is enabled

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/90930
Fixes https://github.com/grafana/grafana/issues/90929

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/core/components/Page/Page.tsx [hunk 1]
```diff
@@ -117,10 +117,19 @@ const getStyles = (theme: GrafanaTheme2) => {
...
+            display: 'flex',
+            flexDirection: 'column',
+            label: 'page-content',
+            flexGrow: 1,
+          }
+        : {
+            label: 'page-content',
+            flexGrow: 1,
+          }
+    ),
```
Comment (joshhunt) [inline public/app/core/components/Page/Page.tsx:132]: Could we tweak this to better show which is different between the two?

```
pageContent: css({
  label: 'page-content',
}, config.featureToggles.bodyScrolling ? { foo } : { blah })
```

**Review by joshhunt — COMMENTED** [review]
Hmm.

This _does_ work, but I get a bit antsy about this needing per-page fixes. Do we reckon there's something we could do in just Page here?

RestoreDashboardsPage needs this fix as well if not.

**Comment by ashharrison90** [comment]
> Hmm.
> 
> This _does_ work, but I get a bit antsy about this needing per-page fixes. Do we reckon there's something we could do in just Page here?
> 
> RestoreDashboardsPage needs this fix as well if not.

i'm not sure we can do anything else tbh... `height: 100%` just won't work correctly anymore in these scenarios. that's what these all boil down to really 😬 

open to suggestions 🙏

**Review by tskarhed — APPROVED** [review]
It works in Safari now, but I do think we need to be a bit prepared for additional complaints.

**Comment by ashharrison90** [comment]
closing in favour of https://github.com/grafana/grafana/pull/91479

**Pull request closed without merging**